integrated_channels/blackboard/client.py

```diff
@@ -40,16 +45,190 @@ def __init__(self, enterprise_configuration):
         self.config = apps.get_app_config('blackboard')
         self.session = None
         self.expires_at = None
+        self.create_course_url = "{}/learn/api/public/v3/courses".format(
+            self.enterprise_configuration.blackboard_base_url
+        )
 
     def create_content_metadata(self, serialized_data):
-        """TODO"""
+        """
+        Create a course from serialized course metadata
+        """
+        channel_metadata_item = json.loads(serialized_data.decode('utf-8'))
+        if 'externalId' not in channel_metadata_item:
+            raise ClientError("No externalId found in metadata, please check json data format", 400)
+
+        external_id = channel_metadata_item['externalId']
+
+        # blackboard does not support all characters in our courseIds so let's gen a hash instead
+        course_id_generated = abs(hash(external_id))
+
+        copy_of_channel_metadata = copy.deepcopy(channel_metadata_item)
```

Comment:
why perform a copy if you're just appending a key that wasn't in there in the first place? (I assume 'courseId' is otherwise not populated)

Reply:
yeah I was just being a good citizen wrt not modifying input vars.. not sure if we utilize this pattern much around our codebase

Reply:
Hmm yeah that's very important in React, but I am not so sure about Python. I'm not 100% up on the best practices though.

I will say that copying variables is definitely not the best practice when dealing with large amounts of data, so it's something to keep in mind. I don't think the data we're dealing with here is particularly large, but I also don't know where this runs precisely and how low of memory limits we may be dealing with.

For instance, in a previous project at a previous company, we were doing a lot of data transformations on csv files, some of which were several gbs in size. Originally because of the different transformations stacked on top of each other, we were copying the original data variable 4+ times. We crashed instances many, many times. These instances had 32+ gb of memory. 😅 

So TL;DR: I don't know if it is normally a best practice or not, but for performance reasons, I'm generally wary of it, but we also probably deal with so little data that it's not a big deal here.

Reply:
yeah totally great points, we have to be watchful of performance when doing this for sure. in this case yeah ti seem innocuous enough to try at least, thanks